yw_website/yw_website/apps/website/views.py

```diff
@@ -107,45 +110,30 @@ def yw_save_ping(request):
     return Response(status=200, data={"data": "You connected to yw web components."})
 
 
+@csrf_exempt
 @api_view(['post'])
 @permission_classes((permissions.AllowAny,))
 def create_workflow(request):
+    #TODO: Replace username with user auth token
     username = User.objects.filter(username=request.data.get('username', None))
+
     if not username:
         return Response(status=500, data={'error': 'bad username'})
-    username = username[0]
-    w = Workflow(
-        user=username,
-        title=request.data.get('title', 'Title'),
-        description=request.data.get('description', 'Description')
-    )
-    w.save()
-    v = Version(
-        workflow=w,
-        yw_model_check_sum=request.data.get('model_checksum', ''),
-        yw_model_output=request.data.get('model', ''),
-        yw_graph_output=request.data.get('graph', ''),
-        last_modified=datetime.datetime.now(tz=timezone.utc)
-    )
-    v.save()
-    r = Run(
-        version=v,
-        yw_recon_output=request.data.get('recon', ''),
-        run_time_stamp=datetime.datetime.now(tz=timezone.utc)
-    )
-    #TODO: Add support for grabbing script and file data
-    r.save()
-    wdata = WorkflowSerializer(w).data
-    wdata['id'] = w.id
-    vdata = VersionSerializer(v).data
-    vdata['id'] = v.id
-    rdata = RunSerializer(r).data
-    rdata['id'] = r.id
-    return Response(status=200, data={
-        "workflow": wdata,
-        "version": vdata,
-        'run': rdata
-    })
+
+    ws = YesWorkflowSaveSerializer(data=request.data)
```

Comment:
this is much more modular two thumbs up :)

Reply:
👍 :D